qiita_db/study.py

```diff
@@ -825,39 +831,58 @@ def ebi_submission_status(self):
         -------
         str
             The study EBI submission status
-        """
-        with qdb.sql_connection.TRN:
-            sql = """SELECT ebi_submission_status
-                     FROM qiita.{0}
-                     WHERE study_id = %s""".format(self._table)
-            qdb.sql_connection.TRN.add(sql, [self.id])
-            return qdb.sql_connection.TRN.execute_fetchlast()
 
-    @ebi_submission_status.setter
-    def ebi_submission_status(self, value):
-        """Sets the study's EBI submission status
-
-        Parameters
-        ----------
-        value : str {%s}
-            The new EBI submission status
-
-        Raises
-        ------
-        ValueError
-            If the status is not known
+        Notes
+        -----
+        There are 4 possible states: 'not submitted', 'submitting',
+        'submitted' & 'failed'. We are going to assume 'not submitted' if the
+        study doesn't have an accession, 'submitted' if it has an accession,
+        'submitting' if there are submit_to_EBI jobs running using the study
+        artifacts, & 'failed' if there are artifacts with failed jobs without
+        successful ones.
         """
-        if not (value in _VALID_EBI_STATUS or
-                value.startswith('failed')):
-            raise ValueError("Unknown status: %s" % value)
+        status = 'not submitted'
         with qdb.sql_connection.TRN:
-            sql = """UPDATE qiita.{}
-                     SET ebi_submission_status = %s
-                     WHERE study_id = %s""".format(self._table)
-            qdb.sql_connection.TRN.add(sql, [value, self.id])
-            qdb.sql_connection.TRN.execute()
-
-    ebi_submission_status.__doc__.format(', '.join(_VALID_EBI_STATUS))
+            if self.ebi_study_accession:
+                status = 'submitted'
```

Comment:
This method should return here, right? If there's an accession there's likely no need for any of the compute done down below.

Reply:
Nop, cause it could be that prep 1 was submitted and the study already has an accession but there is another prep being submitted which should and will overwrite this status.

Reply:
Thanks for the explanation.